lib/metadata/wrapper.js

```diff
@@ -67,54 +67,25 @@ const metadata = {
         });
     },
 
-    putObjectMD: (bucketName, objName, objVal, log, cb, params) => {
+    putObjectMD: (bucketName, objName, objVal, params, log, cb) => {
         log.debug('putting object in metdata');
-        client.putObject(bucketName, objName, objVal, log, err => {
-            if (err) {
```

Comment:
By using the callback `cb` directly we can extend the number of parameters (not limited to `err` and `data`) in the future when MetaData will not inject additional "metadata" (such as `versionId`) directly into the value but return it using another channel.
Also I think it's a good idea not keeping the logging here because, first, this has been logged at bucketclient, and second, different operations may interpret the same error differently, such as `ObjNotFound` for `objectGet` and `objectPut`; these operations should have their own logging.